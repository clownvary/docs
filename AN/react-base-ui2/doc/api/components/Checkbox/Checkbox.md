## Checkbox 
---
UI component of Checkbox

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
disabled|bool|&ensp;|false|Determines enable/disable state.
size|string|&ensp;|false|Determines the Checkbox size.
name|string|&ensp;|false|Field name.
children|node|&ensp;|false|Child nodes.
onChange|func|identity|false|Fires when value change.
value|any|&ensp;|false|Checkbox value.