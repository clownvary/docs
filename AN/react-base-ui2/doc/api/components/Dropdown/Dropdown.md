## Dropdown 
---
UI component that displays Dropdown with variant settings.

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
className|string|&ensp;|false|Custom class name.
style|object|&ensp;|false|Custom style object.
disabled|bool|&ensp;|false|Whether or not to change dropdown.
placeholder|string|'Select one...'|false|Dropdown place holder, and the default value is 'Select one...'.
maxHeight|{<br>&emsp;&emsp;maxHeight: string<br>}|&ensp;|false|The max height of dropdown list.
filter|bool|&ensp;|false|Whether or not to show filter.
filterPlaceholder|string|'Filter...'|false|Filter place holder, and the default value is 'Filter...'.
serverFilter|bool|&ensp;|false|Whether or not to show filter.
autoOpen|bool|false|false|Whether or not to show data list when init.
theme|enum:<br>['flat', 'gradient', 'borderless']|'flat'|false|Dropdown theme such as flat, gradient and borderless. And the default value is flat.
errored|bool|&ensp;|false|Whether or not to show error style.
showCheckbox|bool|&ensp;|false|Whether or not to display checkbox of each option.
showAllCheckbox|bool|&ensp;|false|Whether or not to dispaly check all option.
showTextTip|bool|&ensp;|false|Whether or not to show text tip when hovering the option.
showSpiner|bool|&ensp;|false|Whether or not to show spiner when loading data.
errorInfo|string|&ensp;|false|Error message.
showResults|bool|&ensp;|false|Whether or not to show the count of data list.
results|func|&ensp;|false|Customize how to render Count of data list.
showError|bool|&ensp;|false|Whether or not to show error message.
errorInfoTemplate|string|&ensp;|false|Error infomation template.
onlyDefaultPlaceholder|bool|&ensp;|false|Whether or not to onlyt show text as default place holder.
prefix|string|&ensp;|false|The prefix of text.
value|string &#124; number &#124; object &#124; array|&ensp;|false|Dropdown value.
defaultValue|string &#124; number &#124; object|&ensp;|false|Dropdown default value.
data|[{<br>&emsp;&emsp;text: string,<br>&emsp;&emsp;value: string &#124; number &#124; object<br>}]|[]|false|Display which icon inside of dropdown.
onChange|func|&ensp;|false|Fires when value change.
fuzzyQuery|bool|false|false|Whether or not to get data with fuzzy query.
showTxtOnlyCheckedOneItem|bool|true|false|Whether or not to show checked items text just checked one item.
visible|bool|true|false|Whether or not to show the menu list of Dropdown.
showDeselectall|bool|false|false|Whether or not to show the icon of deselectting all items .
txtSuffix|string|'selected'|false|The suffix text of button.
noResult|string|'No results found.'|false|The text when length of data list is 0.