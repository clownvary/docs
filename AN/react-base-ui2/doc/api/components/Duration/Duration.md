## Duration 
---
UI component of Duration.

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
value|string|&ensp;|false|Sets the duration, the format is 'hh:mm:ss'.
disabled|bool|&ensp;|false|Whether the duration list is disabled.
onChange|func|&ensp;|false|The callback function that is triggered when changes the duration.