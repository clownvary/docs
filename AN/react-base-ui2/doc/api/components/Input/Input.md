## Input 
---
UI component of Input

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
value|string &#124; number|&ensp;|false|Sets the value of the Input.
defaultValue|string &#124; number|&ensp;|false|Sets the default value of the Input.
type|string|'text'|false|Sets the type of the Input and can be 'text', 'password', 'button' etc.
size|enum:<br>['sm', 'md', 'lg']|'md'|false|Determines the Input size.
preIcon|string|&ensp;|false|The label icon displayed before (on the left side of) the input field.
preText|string|&ensp;|false|The label text displayed before (on the left side of) the input field.
postIcon|string|&ensp;|false|The label icon displayed after (on the right side of) the input field.
postText|string|&ensp;|false|The label text displayed after (on the right side of) the input field.
errored|bool|&ensp;|false|Determines if the input has errors.
icon|bool|false|false|Whether add 'input-group--icon' class for the input wrapper.
disabled|bool|&ensp;|false|Whether the input is disabled.
className|string|&ensp;|false|Customize class name for the input wrapper.
style|object|&ensp;|false|Determines the style of the input wrapper.
onChange|func|identity|false|The callback function that is triggered when input value changes.
PreComponent|func &#124; element|&ensp;|false|The component displayed after (on the right side of) the input field.
PostComponent|func &#124; element|&ensp;|false|The component displayed before (on the left side of) the input field.
inputRef|func|identity|false|Gets the instance of the Input Component.
children|node|&ensp;|false|Child Node