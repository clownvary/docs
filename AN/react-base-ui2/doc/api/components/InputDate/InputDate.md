## InputDate  extends InputMoment
---
UI component of InputDate.

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
showTrigger|bool|true|false|Whether to show trigger icon.
showClear|bool|true|false|Whether to show clear icon.