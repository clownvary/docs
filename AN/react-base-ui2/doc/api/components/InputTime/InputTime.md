## InputTime  extends InputMoment
---
InputTime Component

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
showTrigger2|bool|true|false|Icon class name for the 2nd trigger button.
format|string|'t'|false|The format pattern to display the date value. {@link InputMoment.InputMomentPropTypes.format}