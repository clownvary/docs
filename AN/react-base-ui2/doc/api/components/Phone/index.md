## Phone 
---
UI component of Phone Number.

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
value|string|&ensp;|false|The phone number and the format is 'areaCode-mainCode-extendCode'.
disabled|bool|&ensp;|false|Whether the phone field is disabled.
onChange|func|&ensp;|false|The callback function that is triggered when phone field changes.