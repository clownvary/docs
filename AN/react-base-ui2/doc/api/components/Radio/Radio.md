## Radio 
---
UI component of Radio

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
value|string &#124; number|&ensp;|false|Radio value.
disabled|bool|&ensp;|false|Determines enable/disable state.
size|string|&ensp;|false|Determines the Radio size.
name|string|&ensp;|false|Field name.
children|node|&ensp;|false|Child nodes.
onChange|func|identity|false|Fires when value change.