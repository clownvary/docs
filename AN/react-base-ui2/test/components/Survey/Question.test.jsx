import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import moment from 'moment';

import Question from 'src/components/Survey/Question';
import { QuestionType, InputType } from 'src/components/Survey/consts';
import Radio, { RadioGroup } from 'src/components/Radio';
import Checkbox, { CheckboxGroup } from 'src/components/Checkbox';
import Dropdown from 'src/components/Dropdown';
import Phone from 'src/components/Phone';
import InputDate from 'src/components/InputDate';
import InputTime from 'src/components/InputTime';
import Input from 'src/components/Input';
import Duration from 'src/components/Duration';

describe('components/Survey/Question', () => {
  it('Question renders fine', () => {
    const props = {
      isShown: true,
      id: '1',
      text: 'Free to answer something.',
      path: ['snapshot', '1'],
      type: QuestionType.INPUT,
      placeholder: '...',
      format: InputType.FREE,
      hint: 'question hint'
    };

    const snapshot = renderer.create(
      <Question {...props} />
    ).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('Question renders fine with error messages', () => {
    const props = {
      isShown: true,
      id: '2',
      text: 'Say something plz',
      path: ['snapshot', '2'],
      required: true,
      icon: 'icon-check',
      type: QuestionType.INPUT,
      format: InputType.FREE,
      errorMsgs: ['The answer is required.'],
      hint: 'question hint'
    };
    const snapshot = renderer.create(
      <Question {...props} />
    ).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('Question component of Radio type works fine', () => {
    const radioProps = {
      isShown: true,
      id: '3',
      text: 'Do you like drinking coffee?',
      path: ['shallow', '3'],
      type: QuestionType.RADIO,
      hint: 'question hint',
      answers: [
        { text: 'Yes!', value: 'true' },
        { text: 'Nooooo!', value: 'false' }
      ],
      onChange: jest.fn()
    };
    const component = shallow(<Question {...radioProps} />);
    expect(component.find(RadioGroup)).toHaveLength(1);
    const answers = component.find(Radio);
    expect(answers).toHaveLength(2);
    component.find(RadioGroup).simulate('change', { target: { value: 'true' } });
    expect(component.instance().state.value).toBeTruthy();
    expect(radioProps.onChange).toHaveBeenCalled();

    component.setProps({ value: 'false' });
    expect(component.instance().value).toEqual('false');
  });

  it('Question component of Single-Dropdown type works fine', () => {
    const singleDropdownProps = {
      isShown: true,
      id: '3',
      text: 'Do you like drinking coffee?',
      path: ['shallow', '3'],
      type: QuestionType.SINGLEDROPDOWN,
      hint: 'question hint',
      answers: [
        { text: 'Yes!', value: 'true' },
        { text: 'Nooooo!', value: 'false' }
      ],
      onChange: jest.fn()
    };
    const component = shallow(<Question {...singleDropdownProps} />);
    expect(component.find(Dropdown)).toHaveLength(1);

    component.find(Dropdown).simulate('change', { value: 'true' });
    expect(component.instance().state.value).toBeTruthy();
    expect(singleDropdownProps.onChange).toHaveBeenCalled();
  });

  it('Question component of Checkbox type works fine', () => {
    const checkboxProps = {
      isShown: true,
      id: '4',
      text: 'What needs to be added in your coffee?',
      path: ['shallow', '4'],
      type: QuestionType.CHECKBOX,
      hint: 'question hint',
      answers: [
        { text: 'sugar', value: '1' },
        { text: 'milk', value: '2' },
        { text: 'cream', value: '3' }
      ],
      onChange: jest.fn()
    };
    const component = shallow(<Question {...checkboxProps} />);
    expect(component.find(CheckboxGroup)).toHaveLength(1);
    const answers = component.find(Checkbox);
    expect(answers).toHaveLength(3);
    component.find(CheckboxGroup).simulate('change', ['1', '3']);
    expect(component.instance().state.value).toEqual(['1', '3']);
    expect(checkboxProps.onChange).toHaveBeenCalled();
  });

  it('Question component of Multiple-Dropdown type works fine', () => {
    const checkboxProps = {
      isShown: true,
      id: '4',
      text: 'What needs to be added in your coffee?',
      path: ['shallow', '4'],
      type: QuestionType.MULTIPLEDROPDOWN,
      hint: 'question hint',
      answers: [
        { text: 'sugar', value: '1' },
        { text: 'milk', value: '2' },
        { text: 'cream', value: '3' }
      ],
      onChange: jest.fn()
    };
    const component = shallow(<Question {...checkboxProps} />);
    expect(component.find(Dropdown)).toHaveLength(1);
    component.find(Dropdown).simulate('change', { value: ['1', '3'] });
    expect(component.instance().state.value).toEqual(['1', '3']);
    expect(checkboxProps.onChange).toHaveBeenCalled();
  });

  it('Question component of phone type works fine', () => {
    const phoneProps = {
      isShown: true,
      id: '4',
      text: 'What\'s your phone number?',
      path: ['shallow', '4'],
      type: QuestionType.INPUT,
      format: InputType.PHONE,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...phoneProps} />);
    expect(component.find(Phone)).toHaveLength(1);
    expect(component.find(Phone).prop('ariaLabel')).toEqual('What\'s your phone number? question hint');

    component.find(Phone).simulate('change', '12345-234');
    expect(component.instance().state.value).toEqual('12345-234');
    expect(phoneProps.onChange).toHaveBeenCalled();
  });

  it('Question component of phone required is true works fine', () => {
    const phoneProps = {
      isShown: true,
      required: true,
      id: '4',
      text: 'What\'s your phone number?',
      path: ['shallow', '4'],
      type: QuestionType.INPUT,
      format: InputType.PHONE,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...phoneProps} />);
    expect(component.find(Phone).prop('ariaLabel')).toEqual('* What\'s your phone number? (required) question hint');
  });

  it('Question component of date type works fine', () => {
    const dateProps = {
      isShown: true,
      id: '5',
      text: 'When is your birthday?',
      path: ['shallow', '5'],
      type: QuestionType.INPUT,
      format: InputType.DATE,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const birthdayMoment = moment('1983-09-05');
    const component = shallow(<Question {...dateProps} />);
    expect(component.find(InputDate)).toHaveLength(1);

    component.find(InputDate).simulate('valueChange', { value: birthdayMoment });
    expect(component.instance().state.value).toEqual(birthdayMoment.toDate());
    expect(dateProps.onChange).toHaveBeenCalled();
  });

  it('Question component of time type works fine', () => {
    const timeProps = {
      isShown: true,
      id: '6',
      text: 'What\'s the time is it?',
      path: ['shallow', '6'],
      type: QuestionType.INPUT,
      format: InputType.TIME,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const now = moment('2018-08-21 15:51');
    const component = shallow(<Question {...timeProps} />);
    expect(component.find(InputTime)).toHaveLength(1);

    component.find(InputTime).simulate('valueChange', { value: now });
    expect(component.instance().state.value).toEqual(now.toDate());
    expect(timeProps.onChange).toHaveBeenCalled();
  });

  it('Question component of SSN type works fine', () => {
    const ssnProps = {
      isShown: true,
      id: '7',
      text: '???',
      path: ['shallow', '7'],
      type: QuestionType.INPUT,
      format: InputType.SSNTB,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...ssnProps} />);
    expect(component.find(Input)).toHaveLength(1);

    component.find(Input).simulate('change', { target: { value: '661822132' } });
    expect(component.instance().state.value).toEqual('661822132');
    component.find(Input).simulate('blur', { target: { value: '661822132' } });
    expect(ssnProps.onChange).toHaveBeenCalled();
  });

  it('Question component of postal type works fine', () => {
    const postalProps = {
      isShown: true,
      id: '8',
      text: 'What\'s your postal code?',
      path: ['shallow', '8'],
      type: QuestionType.INPUT,
      format: InputType.POSTAL,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...postalProps} />);
    expect(component.find(Input)).toHaveLength(1);

    component.find(Input).simulate('change', { target: { value: '66182-2231' } });
    expect(component.instance().state.value).toEqual('66182-2231');
    component.find(Input).simulate('blur', { target: { value: '66182-2231' } });
    expect(postalProps.onChange).toHaveBeenCalled();
  });

  it('Question component of uppercase type works fine', () => {
    const uppercaseProps = {
      isShown: true,
      id: '9',
      text: 'What\'s your brand name?',
      path: ['shallow', '9'],
      type: QuestionType.INPUT,
      format: InputType.UPPERCASE,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...uppercaseProps} />);
    expect(component.find(Input)).toHaveLength(1);

    component.find(Input).simulate('change', { target: { value: 'aBc' } });
    expect(component.instance().state.value).toEqual('ABC');
    component.find(Input).simulate('blur', { target: { value: 'ABC' } });
    expect(uppercaseProps.onChange).toHaveBeenCalled();
  });

  it('Question component of lowercase type works fine', () => {
    const lowercaseProps = {
      isShown: true,
      id: '10',
      text: 'What\'s your nick name?',
      path: ['shallow', '10'],
      type: QuestionType.INPUT,
      format: InputType.LOWERCASE,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...lowercaseProps} />);
    expect(component.find(Input)).toHaveLength(1);

    component.find(Input).simulate('change', { target: { value: 'Bill' } });
    expect(component.instance().state.value).toEqual('bill');
    component.find(Input).simulate('blur', { target: { value: 'bill' } });
    expect(lowercaseProps.onChange).toHaveBeenCalled();
  });

  it('Question component of number type works fine', () => {
    const numberProps = {
      isShown: true,
      id: '11',
      text: 'How old are you?',
      path: ['shallow', '11'],
      type: QuestionType.INPUT,
      format: InputType.NUMBER,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...numberProps} />);
    expect(component.find(Input)).toHaveLength(1);

    component.find(Input).simulate('change', { target: { value: 35 } });
    expect(component.instance().state.value).toEqual(35);
    component.find(Input).simulate('blur', { target: { value: 35 } });
    expect(numberProps.onChange).toHaveBeenCalled();
  });

  it('Question component of alpha type works fine', () => {
    const alphaProps = {
      isShown: true,
      id: '12',
      text: 'Say something?',
      path: ['shallow', '12'],
      type: QuestionType.INPUT,
      format: InputType.ALPHA,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...alphaProps} />);
    expect(component.find(Input)).toHaveLength(1);

    component.find(Input).simulate('change', { target: { value: 'Ooooops' } });
    expect(component.instance().state.value).toEqual('Ooooops');
    component.find(Input).simulate('blur', { target: { value: 'Ooooops' } });
    expect(alphaProps.onChange).toHaveBeenCalled();
  });

  it('Question component of free type works fine', () => {
    const freeProps = {
      isShown: true,
      id: '12',
      maxLength: 100,
      text: 'Say something?',
      path: ['shallow', '12'],
      type: QuestionType.INPUT,
      format: InputType.FREE,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...freeProps} />);
    expect(component.find(Input)).toHaveLength(1);

    component.find(Input).simulate('change', { target: { value: 'Ooooops' } });
    expect(component.instance().state.value).toEqual('Ooooops');
    component.find(Input).simulate('blur', { target: { value: 'Ooooops' } });
    expect(freeProps.onChange).toHaveBeenCalled();
  });

  it('Question component of duration type works fine', () => {
    const durationProps = {
      isShown: true,
      id: '12',
      text: 'How long do the wind last?',
      path: ['shallow', '13'],
      type: QuestionType.INPUT,
      format: InputType.DURATION,
      answers: [],
      hint: 'question hint',
      onChange: jest.fn()
    };
    const component = shallow(<Question {...durationProps} />);
    expect(component.find(Duration)).toHaveLength(1);

    component.find(Duration).simulate('change', '02:30:00');
    expect(component.instance().state.value).toEqual('02:30:00');
    expect(durationProps.onChange).toHaveBeenCalled();
  });

  it('Question component of free type works fine with clickable icon and no onChange callback', () => {
    const freeProps = {
      isShown: true,
      id: '12',
      maxLength: 100,
      text: 'Say something?',
      path: ['shallow', '12'],
      type: QuestionType.INPUT,
      format: InputType.FREE,
      answers: [],
      icon: 'icon-circle',
      hint: 'question hint',
      onIconClick: jest.fn()
    };
    const component = shallow(<Question {...freeProps} />);
    expect(component.find(Input)).toHaveLength(1);
    component.find(Input).simulate('blur', { target: { value: 'Ooooops' } });

    component.find('span.icon.link').simulate('click');
    expect(freeProps.onIconClick).toHaveBeenCalledTimes(1);

    component.setProps({ required: true });
    component.find('span.icon.link').simulate('click');
    expect(freeProps.onIconClick).toHaveBeenCalledTimes(1);
  });
});
