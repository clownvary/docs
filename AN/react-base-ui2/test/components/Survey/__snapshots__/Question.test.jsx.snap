// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Survey/Question Question renders fine 1`] = `
<div
  className="question aaui-flex"
>
  <div
    className="afx-col question-label afx-xl-6-12"
  >
    <div
      className="question-label__name"
      id="an-question-1_label_id"
    >
      <span
        className="icon icon-circle"
      />
      <span
        className=""
        dangerouslySetInnerHTML={
          Object {
            "__html": "Free to answer something.",
          }
        }
        data-qa-id=""
      />
    </div>
    <div
      className="question-label__hint"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "question hint",
          }
        }
        data-qa-id=""
      />
    </div>
  </div>
  <div
    className="afx-col question-answer-container afx-xl-5-12"
  >
    <div
      className="question-answer"
    >
      <div
        className="input-group input-group--md"
        style={undefined}
      >
        <input
          aria-label="Input box Free to answer something."
          className="input input-group__field input__field"
          disabled={undefined}
          maxLength=""
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          placeholder={undefined}
          type="text"
          value={undefined}
        />
      </div>
    </div>
  </div>
  <div
    className="afx-col afx-xl-1-12 question-icon"
  />
</div>
`;

exports[`components/Survey/Question Question renders fine with error messages 1`] = `
<div
  className="question aaui-flex question-error"
>
  <div
    className="afx-col question-label afx-xl-6-12"
  >
    <div
      className="question-label__name"
      id="an-question-2_label_id"
    >
      <span
        className="icon icon-asterisk"
      />
      <span
        className="question-label__name-required"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Say something plz",
          }
        }
        data-qa-id=""
      />
    </div>
    <div
      className="question-label__hint"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "question hint",
          }
        }
        data-qa-id=""
      />
    </div>
  </div>
  <div
    className="afx-col question-answer-container afx-xl-5-12"
  >
    <div
      className="question-answer"
    >
      <div
        className="input-group input-group--md"
        style={undefined}
      >
        <input
          aria-label="Input box Say something plz"
          className="input input-group__field input__field"
          disabled={undefined}
          maxLength=""
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          placeholder={undefined}
          type="text"
          value={undefined}
        />
      </div>
      <div
        className="question-answer__error"
      >
        <span
          className="icon icon-times-circle"
        />
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "The answer is required.",
            }
          }
          data-qa-id=""
        />
      </div>
    </div>
  </div>
  <div
    className="afx-col afx-xl-1-12 question-icon"
  >
    <span
      className="icon icon-check link is-disabled"
      onClick={[Function]}
    />
  </div>
</div>
`;
